buildSrc/src/main/groovy/org/elasticsearch/gradle/vagrant/VagrantTestPlugin.groovy

```diff
@@ -321,58 +369,169 @@ class VagrantTestPlugin implements Plugin<Project> {
             Task smoke = project.tasks.create("vagrant${boxTask}#smoketest", Exec) {
                 environment vagrantEnvVars
                 dependsOn up
                 finalizedBy halt
-                commandLine 'vagrant', 'ssh', box, '--command',
-                        "set -o pipefail && echo 'Hello from ${project.path}' | sed -ue 's/^/    ${box}: /'"
+                description "Tries to run a 'hello world' shell command on box ${box}"
+            }
+            if (box in LINUX_BOXES) {
+                smoke.commandLine = ['vagrant', 'ssh', box, '--command', """
+                    set -euo pipefail
+                    echo 'Hello from ${project.path}' | sed -ue 's/^/    ${box}: /'
+                """]
+            } else {
+                smoke.commandLine = ['vagrant', 'winrm', box, '--command', wrapPowershell("""
+                    Write-Host '    ${box}: Hello from ${project.path}'
+                """)]
             }
             vagrantSmokeTest.dependsOn(smoke)
 
-            Task packaging = project.tasks.create("vagrant${boxTask}#packagingTest", BatsOverVagrantTask) {
-                remoteCommand BATS_TEST_COMMAND
+            Task relocateProject = project.tasks.create("vagrant${boxTask}#relocateProject", VagrantCommandTask) {
                 boxName box
                 environmentVars vagrantEnvVars
-                dependsOn up, setupBats
+                dependsOn up
                 finalizedBy halt
+                description "Run a command on box ${box} to move the project to vagrant's home directory"
             }
+            if (box in LINUX_BOXES) {
+                relocateProject.command = 'ssh'
+                relocateProject.args = ['--command', """
+                    set -euo pipefail
+                    rm -rf ~/elasticsearch ~/elasticsearch-extra
+                    rsync -rlt /elasticsearch/ ~/elasticsearch
+                    if [ -d /elasticsearch-extra ]; then
+                        rsync -rlt /elasticsearch-extra/ ~/elasticsearch-extra
+                    fi
+                """]
+            } else {
+                relocateProject.command = 'winrm'
+                relocateProject.args = ['--command', wrapPowershell("""
+                    if (Test-Path "~/elasticsearch") {
+                        Remove-Long-Path C:\\Users\\vagrant\\elasticsearch
+                    }
+                    if (Test-Path "~/elasticsearch-extra") {
+                        Remove-Long-Path C:\\Users\\vagrant\\elasticsearch-extra
+                    }
 
-            TaskExecutionAdapter packagingReproListener = new TaskExecutionAdapter() {
+                    Copy-Long-Path C:\\elasticsearch C:\\Users\\vagrant\\elasticsearch
+
+                    if (Test-Path "C:/elasticsearch-extra") {
+                        Copy-Long-Path C:\\elasticsearch-extra C:\\Users\\vagrant\\elasticsearch-extra
+                    }
+                """)]
+            }
```

Comment:
It's worth noting that this task is very slow, on the order of minutes. It seems like it's necessary on windows because the project is mounted in such a way that gradle has trouble running in it - I wasn't able to figure out a workaround. On linux, we can probably avoid it if that's too time consuming